contracts/DutchExchange.sol

```diff
@@ -320,51 +303,58 @@ contract DutchExchange {
     )
         public
     {
-
-        // Amount mmust be > 0
         amount = Math.min(amount, balances[sellToken][msg.sender]);
+        // R1: amount mmust be > 0
         require(amount > 0);
+        uint latestAuctionIndex = getAuctionIndex(sellToken, buyToken);
+        uint auctionStart = getAuctionStart(sellToken, buyToken);
+
+
+        uint closingPriceDen = closingPrices[sellToken][buyToken][auctionIndex - 1].den;
+        uint closingPriceDenOpp = closingPrices[buyToken][sellToken][auctionIndex - 1].den;
 
-        //uint latestAuctionIndex = latestAuctionIndices[token1][token2];
-        if (now < auctionStarts[orderedPairFristElement(buyToken, sellToken)][orderedPairSecondElement(buyToken, sellToken)]) {
-            // We are in the 10 minute buffer period (or 6 hours for new token pair)
+        if (now < auctionStart) {
+            // C1: We are in the 10 minute buffer period
             // Auction has already cleared, and index has been incremented
-            // Sell order must use that auction index
-            require(auctionIndex == latestAuctionIndices[orderedPairFristElement(buyToken, sellToken)][orderedPairSecondElement(buyToken, sellToken)]);
+            // R1.1: sell order must use that auction index
+            require(auctionIndex == latestAuctionIndex);
         } else {
-            // check whether current auchtion has no starting date, because auctions have not been funded.
-            if (auctionIndex == latestAuctionIndices[orderedPairFristElement(buyToken, sellToken)][orderedPairSecondElement(buyToken, sellToken)]){
-                  require ( auctionStarts[orderedPairFristElement(buyToken, sellToken)][orderedPairSecondElement(buyToken, sellToken)] == 1);
+
+
+            // C2
+            // R2.1: sell orders must go to next auction
+            if (getAuctionStart(sellToken, buyToken) < now && closingPriceDen > 0 && closingPriceDenOpp > 0){
```

Comment:
Currently `getAuctionStart(sellToken, buyToken) < now` is always true at least for the first, freshly created toke pair, so I can repeatedly post sell orders to the ongoing auction.

Reply:
solved this one in a seperate commit